drivers/metadata/mysql/metadata.go

```diff
@@ -0,0 +1,58 @@
+package mysql
+
+import (
+	"time"
+
+	"github.com/gohxs/readline"
+	"github.com/xo/usql/drivers"
+	"github.com/xo/usql/drivers/completer"
+	"github.com/xo/usql/drivers/metadata"
+	infos "github.com/xo/usql/drivers/metadata/informationschema"
+)
+
+var (
+	// NewReader for MySQL databases
+	NewReader = infos.New(
+		infos.WithPlaceholder(func(int) string { return "?" }),
+		infos.WithSequences(false),
+		infos.WithCheckConstraints(false),
+		infos.WithCustomClauses(map[infos.ClauseName]string{
+			infos.ColumnsNumericPrecRadix:         "10",
+			infos.FunctionColumnsNumericPrecRadix: "10",
+			infos.ConstraintIsDeferrable:          "''",
+			infos.ConstraintInitiallyDeferred:     "''",
+			infos.ConstraintJoinCond:              "AND r.table_name = f.table_name",
+		}),
+		infos.WithSystemSchemas([]string{"mysql", "information_schema", "performance_schema", "sys"}),
+		infos.WithCurrentSchema("COALESCE(DATABASE(), '%')"),
+	)
+	// NewCompleter for MySQL databases
+	NewCompleter = func(db drivers.DB, opts ...completer.Option) readline.AutoCompleter {
+		readerOpts := []metadata.ReaderOption{
+			// this needs to be relatively low, since autocomplete is very interactive
+			metadata.WithTimeout(3 * time.Second),
+			metadata.WithLimit(1000),
+		}
+		reader := NewReader(db, readerOpts...)
+		opts = append([]completer.Option{
+			completer.WithReader(reader),
+			completer.WithDB(db),
+			completer.WithSQLStartCommands(append(completer.CommonSqlStartCommands, "USE")),
+			completer.WithBeforeComplete(func(previousWords []string, text []rune) [][]rune {
+				schemaNames := []string{}
+				schemas, err := reader.(metadata.SchemaReader).Schemas(metadata.Filter{WithSystem: true})
```

Comment:
This should only be done either when matched `USE` or before creating the completer, because otherwise a query fetching schemas/databases will be executing _on each tab key press_.